net-misc/asterisk-base/files/initd-3

```diff
@@ -0,0 +1,313 @@
+		OPTS="${OPTS} -g"
+
+		if [ -n "${ASTERISK_CORE_DIR}" ] && \
+		   [ ! -d "${ASTERISK_CORE_DIR}" ]
+		then
+			checkpath -d -m 0755 -o ${ast_user}:${ast_group} "${ASTERISK_CORE_DIR}"
+		fi
+		ASTERISK_CORE_DIR="${ASTERISK_CORE_DIR:-/tmp}"
+
+		cd "${ASTERISK_CORE_DIR}"
+		einfo "Core dump size            : ${ASTERISK_CORE_SIZE}"
+		einfo "Core dump location        : ${ASTERISK_CORE_DIR}"
+
+		if [ -n "${ASTERISK_CORE_COMPRESS}" ]; then
+			einfo "Core compression tool     : ${ASTERISK_CORE_COMPRESS}"
+			WRAPPER_OPTS="${WRAPPER_OPTS} --corecompress ${ASTERISK_CORE_COMPRESS}"
+		fi
+
+		if yesno "${ASTERISK_CORE_USEHOSTNAME}"; then
+			WRAPPER_OPTS="${WRAPPER_OPTS} --corerename core-%H-%D-%T"
+		else
+			WRAPPER_OPTS="${WRAPPER_OPTS} --corerename core-%D-%T"
+		fi
+	else
+		cd /
+	fi
+
+	if [ -n "${ASTERISK_MAX_FD}" ]; then
+		WRAPPER_OPTS="${WRAPPER_OPTS} --maxfd ${ASTERISK_MAX_FD}"
+		einfo "Max open filedescriptors  : ${ASTERISK_MAX_FD}"
+	fi
+
+	if [ -n "${ASTERISK_NICE}" ]; then
+		if [ ${ASTERISK_NICE} -ge -20 ] && \
+		   [ ${ASTERISK_NICE} -le  19 ]; then
+			einfo "Nice level                : ${ASTERISK_NICE}"
+			WRAPPER_OPTS="${WRAPPER_OPTS} --nicelevel ${ASTERISK_NICE}"
+		else
+			eerror "Nice value must be between -20 and 19"
+			return 1
+		fi
+	fi
+
+	if [ -n "${ASTERISK_NOTIFY_EMAIL}" ]; then
+		if [ -x /usr/sbin/sendmail ]; then
+			einfo "Email notifications go to : ${ASTERISK_NOTIFY_EMAIL}"
+			WRAPPER_OPTS="${WRAPPER_OPTS} --crashemail ${ASTERISK_NOTIFY_EMAIL}"
+		else
+			ewarn "Notifications disabled, /usr/sbin/sendmail doesn't exist or is not executable!"
+		fi
+	fi
+
+	if [ -n "${ASTERISK_TTY}" ]; then
+		for x in "${ASTERISK_TTY}" \
+			 "/dev/tty${ASTERISK_TTY}" \
+			 "/dev/vc/${ASTERISK_TTY}"
+		do
+			if [ -c "${x}" ]; then
+				TTY="${x}"
+			fi
+		done
+		if [ -n "${TTY}" ]; then
+			einfo "Messages are sent to      : ${TTY}"
+			WRAPPER_OPTS="${WRAPPER_OPTS} --tty ${TTY}"
+		else
+			ewarn "ASTERISK_TTY=${TTY} ignored, not valid."
+		fi
+	fi
+
+	if yesno "${ASTERISK_CONSOLE}" && [ -n "${TTY}" ]; then
+		einfo "Starting Asterisk console : Yes"
+		OPTS="${OPTS} -c"
+	fi
+
+	if ! getent passwd "${ast_user}" >/dev/null 2>&1; then
+		eerror "Requested to run asterisk as ${ast_user}, which doesn't exist."
+		return 1
+	fi
+	OPTS="${OPTS} -U ${ast_user}"
+
+	if [ -n "${ast_group}" ] && ! getent group "${ast_group}" >/dev/null 2>&1; then
+		eerror "Requested to run ${ast_instancename} with group ${ast_group}, which doesn't exist."
+		return 1
+	fi
+	[ -n "${ast_group}" ] && OPTS="${OPTS} -G ${ast_group}"
+
+	if [ "${ast_user}" = root ]; then
+		ewarn "Starting asterisk as root is not recommended (SERIOUS SECURITY CONSIDERATIONS)."
+	elif [ "${ast_group}" = root ]; then
+		ewarn "Starting asterisk with group root is not recommended (SERIOUS SECURITY CONSIDERATIONS)."
+	fi
+
+	checkpath -d -m 0755 -o "${ast_user}:${ast_group}" "${ast_logdir}" "${ast_rundir}"
```

Comment:
I think these two lines should come with big warnings in the conf file. If the user sets (say) `ASTERISK_LOGDIR=/var/log`, this line is going to make `/var/log/` writable by the asterisk user.

(IMO some things just shouldn't be configurable, but we don't need to rewrite the entire script today.)

Reply:
config path being configurable is critical for us.

We don't run just one asterisk instance on a server, most insane we've done to date is slightly over 100.  Each really should have it's own logfile.  You are right about this specific use case, perhaps this as an alternative:

If the folder exist (prior to invoking checkpath) and is at that point does NOT have the appropriate ownership, refuse to start?

Reply:
These aren't currently mentioned in the confd file at all, and I just verified that we depend on the automatically *calculated based on RC_SVCNAME* value.  Added an ewarn to initd if ASTERISK_{RUNDIR,LOGDIR} is set before honouring it anyway.  Depending on how long before v4 I might turn it into eerror and failing then.

https://github.com/jkroonza/asterisk-base/commit/03c7266b163fa2e9cca1ef0de6077be75d1235a4

Reply:
Yeah I wasn't suggesting a single hard-coded value, but something like `/var/log/${RC_SVCNAME}`. That lets you create as many instances as you want but also absolves the init script of the responsibility to handle all the stupid things that the user might do.